dragonfly/block/fence_gate.go

```diff
@@ -0,0 +1,135 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/block/model"
+	"github.com/df-mc/dragonfly/dragonfly/block/wood"
+	"github.com/df-mc/dragonfly/dragonfly/item"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"github.com/df-mc/dragonfly/dragonfly/world/sound"
+	"github.com/go-gl/mathgl/mgl64"
+)
+
+// FenceGate is a block that can be used as an openable 1x1 barrier.
+type FenceGate struct {
+	noNBT
+	transparent
+
+	// Wood is the type of wood of the fence gate. This field must have one of the values found in the material
+	// package.
+	Wood wood.Wood
+	// Facing is the direction the fence gate swings open.
+	Facing world.Direction
+	// Open is whether the fence gate is open.
+	Open bool
+	// InWall lowers the fence gate by 3 pixels if placed next to wall blocks.
+	InWall bool
```

Comment:
Might be better to call this 'Lowered' and document that it happens when placed next to walls.